exonum-light-client/src/main/java/com/exonum/client/ExonumClient.java

```diff
@@ -77,6 +80,91 @@
    */
   Optional<TransactionResponse> getTransaction(HashCode id);
 
+  /**
+   * Returns the <em>blockchain height</em> which is the height of the latest committed block
+   * in the blockchain. The block height is a distance between the last block
+   * and the "genesis", or initial, block. Therefore, the blockchain height is equal to the number
+   * of blocks plus one.
+   *
+   * @throws RuntimeException if the client is unable to complete a request
+   *        (e.g., in case of connectivity problems)
+   */
+  long getBlockchainHeight();
+
+  /**
+   * Returns the information about the block with transaction hashes included at this block.
+   * @param height blockchain height starting from 0 (genesis block)
+   * @return block information response
+   * @throws RuntimeException if block is not found by the requested height,
+   *        i.e. the requested height is greater than actual blockchain height
+   * @throws IllegalArgumentException if the given height is negative
+   * @throws RuntimeException if the client is unable to complete a request
+   *        (e.g., in case of connectivity problems)
+   */
+  BlockResponse getBlockByHeight(long height);
+
+  /**
+   * Returns blockchain blocks information for the requested range.
+   * @param count Number of blocks to return.
+   *        It should be in range [1, {@linkplain ExonumApi#MAX_BLOCKS_PER_REQUEST}]
+   * @param skipEmpty if {@code true}, then only non-empty blocks will be returned
+   * @param heightMax maximum height of the returned blocks. The blocks are returned
+   *        in reverse order, starting from the {@code heightMax}
+   *        and up to the {@code heightMax - count + 1}.
```

Comment:
Now it is not correct (they might get below heightMax - count + 1 if empty blocks are involved). Shall we remove that?

Reply:
also thought about that